### PR TITLE
Fix yaml serializer test

### PR DESCRIPTION
**Pull request by benediktwerner**
### What does this PR do?
Fixes the `unit.serializers.test_serializers.TestSerializers.test_serialize_yaml` test from failing and replaces all the `assert` statements in the same file with the corresponding `TestCase.assert` functions.

### What issues does this PR fix or reference?
new behaviour/test was introduces by #41956

### Previous Behavior
#41956 added a custom tag to the yaml serializer to mark encrypted strings. The corresponding test fails because of missing quotation marks.

The failing assert statement is in line 51. The expected result is `{encrypted_data: !encrypted foo, foo: bar}` but the real result is `{encrypted_data: !encrypted 'foo', foo: bar}`.

As far as I can tell the real result is how it should work based on the YAML specification.

As the test seems to have passed for other people before I'm not sure if maybe the PyYAML implementation is different on some operating systems? I tested the following code on `Ubuntu 16.04 LTS` with `Python 2.7.12` and on `Windows 10` with `Python 3.5.2`:
```python
import yaml
class EncryptedString(str):
    @staticmethod
    def yaml_dumper(dumper, data):
        return dumper.represent_scalar(u"!encrypted", data.__str__())
yaml.SafeDumper.add_multi_representer(EncryptedString, EncryptedString.yaml_dumper)
print(yaml.dump({"encrypted_data": EncryptedString("foo")}, Dumper=yaml.SafeDumper))
```
which mimics the behavior of the Salt YAML parser and it always produced `{encrypted_data: !encrypted 'foo'}` with quotation marks.

### New Behavior
Changed expected value to match real value and replaced assert statements.

### CODE REVIEW

File: tests/unit/serializers/test_serializers.py [hunk 1]
```diff
@@ -48,21 +48,21 @@ def test_serialize_yaml(self):
             "encrypted_data": EncryptedString("foo")
         }
         serialized = yaml.serialize(data)
-        assert serialized == '{encrypted_data: !encrypted foo, foo: bar}', serialized
+        self.assertEqual(serialized, '{encrypted_data: !encrypted \'foo\', foo: bar}')
```
Comment (benediktwerner) [inline tests/unit/serializers/test_serializers.py]: This is the important change: `foo` --> `\'foo\'`

Comment (benediktwerner) [inline tests/unit/serializers/test_serializers.py]: This is the important change: `foo` --> `\'foo\'`

**Comment by cachedout** [comment]
@yagnik and @terminalmage Any thoughts here?

**Comment by yagnik** [comment]
I want to say that we should only support one style i.e. the one written in the spec but can only say that once we have a concrete reason as to where this difference is. I'll try to debug too and report back.

**Comment by cachedout** [comment]
@yagnik Sounds good. Keep us posted. Thanks.

**Comment by cachedout** [comment]
@yagnik Any updates here?

**Comment by yagnik** [comment]
Go Go Jenkins

**Comment by yagnik** [comment]
Look ma I found something!
Sorry I didn't update earlier.

@benediktwerner is correct that pyyaml requires quotations, the issue comes when salt finds c bindings on the system here: https://github.com/saltstack/salt/blob/7c0398a8bdf949b3126b33a5b2d2bfaf85b47df2/salt/serializers/yaml.py#L28-L29

CSafedumper doesn't like to put quotations whereas default pyyaml does.
In terms of loading both are nice and treat as a string.

To prove the issue:
```
import yaml
class EncryptedString(str):
    @staticmethod
    def yaml_dumper(dumper, data):
        return dumper.represent_scalar(u"!encrypted", data.__str__())

yaml.SafeDumper.add_multi_representer(EncryptedString, EncryptedString.yaml_dumper)
print(yaml.dump({"encrypted_data": EncryptedString("foo")}, Dumper=yaml.SafeDumper))
yaml.CSafeDumper.add_multi_representer(EncryptedString, EncryptedString.yaml_dumper)
print(yaml.dump({"encrypted_data": EncryptedString("foo")}, Dumper=yaml.CSafeDumper))

{encrypted_data: !encrypted 'foo'}

{encrypted_data: !encrypted foo}

```

Going by the yaml spec http://www.yaml.org/spec/1.2/spec.html#id2761292 for explicit tags, it shows that quotations aren't present so I want to say that pyyaml is the one at fault.

I'll open an issue on pyyaml to see what the authors think about this.

**Comment by rallytime** [comment]
@benediktwerner This PR has a merge conflict now. However, @yagnik @terminalmage I know you both have been involved in this space recently. What are your thoughts on this PR?

**Comment by benediktwerner** [comment]
I guess the test really should stay as it is and PyYaml should fix the quotations. The question is how the test should work until that is done. Should it simply accept both styles or check which parser is used or just fail? Or maybe get skipped? I think the behavior should at least be consistent on any system.

In any case, it's probably still a good idea to replace the `assert` statements with the correct `assert` functions. If you want I can also split those changes out into another pull request and fix the merge conflict.

**Comment by terminalmage** [comment]
The tests aren't failing now, they appear to have been fixed in an unrelated PR, likely during the unicode work we were doing. I think we can close this.

**Comment by benediktwerner** [comment]
Are you sure they pass on all systems? I don't think any unicode changes can fix this and the tests always pass on systems with c-bindings for the yaml parser. They only fail when the PyYaml parser is used, because that parser always adds quotation marks to tags.

**Comment by terminalmage** [comment]
They're passing on all the platforms we test, my guess is that older platforms which don't have a CSafeDumper are the only ones failing. In that case we should probably just skip the test `if not hasattr(yaml, 'CSafeDumper')`.

**Comment by benediktwerner** [comment]
Okay, I just checked again and realized you changed the test when fixing the unicode stuff in commit [469d8e6e](https://github.com/saltstack/salt/commit/469d8e6e52501c4f055c475b68ba804513935653), which was part of #45438. Now the expected value depends on the active parser. Not really an ideal solution but I guess it works? It's probably not worth putting too much work into this, so maybe that's the way to go for now. I fixed the merge conflict so all this PR now does is fix the use of `assert`-statements.

**Comment by terminalmage** [comment]
I think you meant https://github.com/saltstack/salt/pull/45658?

**Comment by terminalmage** [comment]
@rallytime are we concerned with the use of `assert` vs. the unittest2-specific asserts, given our desire to eventually move to pytest?

**Comment by rallytime** [comment]
@terminalmage No, not particularly. The test suite works with pytest-style asserts as well as the unittest type of asserts.

**Comment by terminalmage** [comment]
@rallytime OK, in that case, maybe this PR is not necessary since the end goal is to make these tests pytest-compatible. What do you think?

**Comment by rallytime** [comment]
@terminalmage Yes, I think you're correct. Apologies for the long wait and back and forth here @benediktwerner, but since we'll be moving towards pytest soon (and both pytest and unittest are supported now in our test set up), I am going to close this PR. Thanks again for your contributions.

**Pull request closed without merging**